code/modules/recycling/disposal-unit.dm

```diff
@@ -527,6 +527,93 @@
 		M.loc = src
 	flush()
 
+
+//Trap Door
+obj/machinery/disposal/trapdoor
+	name = "trapdoor"
+	desc = "A chute for big and small packages alike!"
+	density = 0
+	icon = 'icons/obj/atmospherics/pipes/trapdoor.dmi'
+	icon_state = "closed"
+	mode = 0 // the chute doesn't need charging and always works
+	layer = 2.01
+	var/trap_door_state = 1
+	var/trap_closed = 1
+	var/id = 1
+	var/auto_close = 1200
+	var/sound_open = 'sound/machines/blast_door.ogg'
+	var/sound_close = 'sound/machines/blast_door.ogg'
+	var/operating = 0
+
+
+	process()
```

Comment:
This is a different method of defining procs, but it'd be better to just use the one used pratically everywhere aka /obj/machinery/disposal/trapdoor/process()